python/ray/_private/function_manager.py

```diff
@@ -520,28 +530,17 @@ def _load_actor_class_from_gcs(self, job_id,
         try:
             with self.lock:
                 actor_class = pickle.loads(pickled_class)
+            assert False
         except Exception:
-            logger.exception("Failed to load actor class %s.", class_name)
+            logger.debug("Failed to load actor class %s.", class_name)
+            # If an exception was thrown when the actor was imported, we record
+            # the traceback and notify the scheduler of the failure.
+            traceback_str = format_error_message(traceback.format_exc())
             # The actor class failed to be unpickled, create a fake actor
             # class instead (just to produce error messages and to prevent
             # the driver from hanging).
             actor_class = self._create_fake_actor_class(
-                class_name, actor_method_names)
-            # If an exception was thrown when the actor was imported, we record
-            # the traceback and notify the scheduler of the failure.
-            traceback_str = ray._private.utils.format_error_message(
-                traceback.format_exc())
-            # Log the error message.
-            push_error_to_driver(
-                self._worker,
-                ray_constants.REGISTER_ACTOR_PUSH_ERROR,
-                f"Failed to unpickle actor class '{class_name}' "
-                f"for actor ID {self._worker.actor_id.hex()}. "
-                f"Traceback:\n{traceback_str}",
-                job_id=job_id)
-            # TODO(rkn): In the future, it might make sense to have the worker
-            # exit here. However, currently that would lead to hanging if
-            # someone calls ray.get on a method invoked on the actor.
+                class_name, actor_method_names, traceback_str)
```

Comment:
Replaces previous separate errors with unified reporting in RuntimeException above.